### PR TITLE
Add ineffassign test

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

* Add ineffassign test 
* Fix existing ineffectual assignments

### CODE REVIEW

File: pkg/metrics/metric.go [hunk 1]
```diff
@@ -44,15 +44,6 @@ func (a APIMetricType) MarshalText() ([]byte, error) {
 
 // UnmarshalText is a custom unmarshaller for APIMetricType (used for testing)
 func (a APIMetricType) UnmarshalText(buf []byte) error {
-	tmp := string(buf)
```
Comment (mfpierre) [inline pkg/metrics/metric.go]: Not sure what this function is supposed to do, but removing it crashes some tests

File: pkg/util/pipe_windows.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ func (p *WinNamedPipe) Open() error {
 		return nil
 	}
 
-	c, err := winio.DialPipe(p.path, nil)
+	_, err := winio.DialPipe(p.path, nil)
```
Comment (mfpierre) [inline pkg/util/pipe_windows.go]: I find this one weird as well

File: pkg/util/docker/docker_util.go [hunk 3]
```diff
@@ -235,7 +235,6 @@ func (d *dockerUtil) containers(cfg *ContainerListConfig) ([]*Container, error)
 		containers, ok = cached.([]*Container)
 		if !ok {
 			log.Errorf("invalid cache format, forcing a cache miss")
-			hit = false
```
Comment (mfpierre) [inline pkg/util/docker/docker_util.go]: Not sure what was the expected outcome there? (hit is not reused bellow)

File: pkg/metrics/metric.go (hunk 1 above)
Comment (hkaj) [reply pkg/metrics/metric.go]: cc @gmmeyer can you provide more context? 🙏

File: pkg/util/docker/docker_util.go (hunk 3 above)
Comment (hkaj) [reply pkg/util/docker/docker_util.go]: That's likely a bug. https://github.com/DataDog/datadog-agent/blob/9a735338a8c094478225860b8a371c194d029b7e/pkg/util/docker/docker_util.go#L240 This should be `if !hit`, not `else` I think.

File: pkg/util/pipe_windows.go [hunk 4]
```diff
@@ -40,7 +40,8 @@ func (p *WinNamedPipe) Open() error {
 		defer l.Close()
 	}
 
-	if c, err = winio.DialPipe(p.path, nil); err != nil {
+	c, err := winio.DialPipe(p.path, nil)
+	if err != nil {
```
Comment (hkaj) [inline pkg/util/pipe_windows.go]: 👍

**Review by hkaj — COMMENTED** [review]
This is great 🍪 
Let's wait on details for the Unmarshal thing, but everything else looks good to me.

**Review by masci — APPROVED** [review]

**Pull request merged**